### PR TITLE
Allow filter parameter for ACL do we can get permission from JSON request

### PR DESCRIPTION
**Pull request by gabrielfs7**
Allow filter parameter for ACL do we can get permission from JSON request

https://oat-sa.atlassian.net/browse/ADF-8

PRS related: 

- [ ] https://github.com/oat-sa/tao-core/pull/2874
- [ ] https://github.com/oat-sa/extension-tao-mediamanager/pull/264/files
- [ ] https://github.com/oat-sa/extension-tao-dac-simple/pull/162

### CODE REVIEW

File: models/classes/AdvancedSearch/AdvancedSearchChecker.php [hunk 1]
```diff
@@ -40,6 +40,8 @@ public function isEnabled(): bool
 
     private function isAdvancedSearchEnabled(): bool
     {
+        return false;
```
Comment (gabrielfs7) [inline models/classes/AdvancedSearch/AdvancedSearchChecker.php:43]: I need to remove it after tests

**Review by boajer — APPROVED** [review]
Backend part
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: models/classes/accessControl/filter/FormParameterFilter.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+            if (in_array($key, $filterNames, true) && common_Utils::isUri($encodedUri)) {
+                $groupedUris[$key][] = $encodedUri;
+            }
+        }
+
+        return $groupedUris;
+    }
+
+    private function flattenArray(array $multiDimensionalArray)
```
Comment (shpran) [inline models/classes/accessControl/filter/FormParameterFilter.php:50]: Can we declare return type?

File: models/classes/accessControl/filter/ParameterFilterProxy.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+{
+    /** @var ParameterFilterInterface */
+    private $filters;
+
+    public function __construct(ParameterFilterInterface ...$filters)
+    {
+        $this->filters = empty($filters) ? [
+            new FormParameterFilter(),
+            new JsonParameterFilter(),
+        ] : $filters;
```
Comment (shpran) [inline models/classes/accessControl/filter/ParameterFilterProxy.php:34]:
```suggestion
        $this->filters = $filters ?: [
            new FormParameterFilter(),
            new JsonParameterFilter(),
        ];
```

File: test/unit/models/classes/accessControl/filter/FormParameterFilterTest.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
+<?php declare(strict_types=1);
```
Comment (shpran) [inline test/unit/models/classes/accessControl/filter/FormParameterFilterTest.php:1]: Maybe I'm wrong, but we declare `strict_type` after the dock block and before the namespace

**Review by shpran — APPROVED** [review]
Good job!
Left several comments. Please, don't forget to declare `strict_types` for new files 👍🏻 

- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: test/unit/models/classes/accessControl/filter/FormParameterFilterTest.php (hunk 4 above)
Comment (gabrielfs7) [reply test/unit/models/classes/accessControl/filter/FormParameterFilterTest.php]: You are right, fixed!

File: models/classes/accessControl/filter/FormParameterFilter.php (hunk 2 above)
Comment (gabrielfs7) [reply models/classes/accessControl/filter/FormParameterFilter.php]: true, done! :)

**Review by boajer — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by habemuscode — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**